### PR TITLE
Upload tvOS TestFlight builds correctly

### LINKED ISSUE #55: Unable to upload tvOS apps
I used gym to build a tvOS app (incidentally, I had to use `--destination generic/platform=tvOS` to convince gym that it was not building an iOS app).

I then used pilot to upload the app, but got this error:

    Return status of iTunes Transporter was 1: ERROR ITMS-90545: "Invalid Provisioning Profile. This provisioning profile is not compatible with iOS apps."

I'm able upload the app just fine using the Organizer, but would prefer to automate this.

### PR DESCRIPTION
**Pull request by tmm1**
Same concept as https://github.com/fastlane/fastlane_core/pull/58

fixes #55 and #92

### CODE REVIEW

File: lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -9,7 +9,8 @@ def upload(options)
 
       package_path = PackageBuilder.new.generate(apple_id: app.apple_id,
                                                  ipa_path: config[:ipa],
-                                             package_path: "/tmp")
+                                             package_path: "/tmp",
+                                                 platform: "appletvos")
```
Comment (tmm1) [inline lib/pilot/build_manager.rb]: This should obviously not be hardcoded, but I'm not sure what the best way to detect platform for the given ipa or app is.

**Comment by tmm1** [comment]
Got past the upload now, but still fails to distribute the new build:

```
[03:00:56]: Successfully uploaded the new binary to iTunes Connect
[03:00:56]: Waiting for iTunes Connect to process the new build
[03:01:33]: Waiting for iTunes Connect to process the new build
[03:02:08]: Waiting for iTunes Connect to process the new build
[03:02:45]: Waiting for iTunes Connect to process the new build
[03:03:21]: Waiting for iTunes Connect to process the new build
[03:04:01]: Waiting for iTunes Connect to finish processing the new build (1.0.0 - 1)
[03:04:31]: Successfully finished processing the build
[03:04:31]: You can now tweet:
[03:04:31]: iTunes Connect #iosprocessingtime 4 minutes
[03:04:31]: Distributing new build to testers
gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:207:in `handle_itc_response': unable_to_get_build (Spaceship::TunesClient::ITunesConnectError)
	from gems/spaceship-0.19.3/lib/spaceship/tunes/tunes_client.rb:551:in `update_build_information!'
	from gems/spaceship-0.19.3/lib/spaceship/tunes/build.rb:127:in `update_build_information!'
	from bundler/gems/pilot-5384247f2789/lib/pilot/build_manager.rb:107:in `distribute_build'
	from bundler/gems/pilot-5384247f2789/lib/pilot/build_manager.rb:23:in `upload'
	from bundler/gems/pilot-5384247f2789/lib/pilot/commands_generator.rb:48:in `block (2 levels) in run'
```

**Comment by tmm1** [comment]
This now works after https://github.com/fastlane/spaceship/pull/230

**Comment by KrauseFx** [comment]
Instead of applying the same change in this repo, I'd much rather prefer updating `pilot` to use the `ipa_upload_package_builder` class from `fastlane_core`. Would you mind updating the PR to do that? :+1:

File: lib/pilot/build_manager.rb [hunk 2]
```diff
@@ -7,9 +7,10 @@ def upload(options)
 
       Helper.log.info "Ready to upload new build to TestFlight (App: #{app.apple_id})...".green
 
-      package_path = PackageBuilder.new.generate(apple_id: app.apple_id,
-                                                 ipa_path: config[:ipa],
-                                             package_path: "/tmp")
+      package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(app_id: app.apple_id,
+                                                                      ipa_path: config[:ipa],
+                                                                  package_path: "/tmp",
+                                                                      platform: "appletvos")
```
Comment (tmm1) [inline lib/pilot/build_manager.rb]: I think there's some sort of ipa plist parser available that we might be able to use here? Need to figure out what the platform for the ipa is.

Comment (KrauseFx) [inline lib/pilot/build_manager.rb]: Is the tvOS binary always separate from the iOS one? How is the binary different? If we can fetch it from the plist we should be able to do that:
```
FastlaneCore::IpaFileAnalyser.fetch_info_plist_file
```

**Comment by tmm1** [comment]
@KrauseFx I used IpaFileAnalyser like you suggested and this works great now. Please merge =)

File: lib/pilot/build_manager.rb [hunk 3]
```diff
@@ -7,9 +7,11 @@ def upload(options)
...
       Helper.log.info "Ready to upload new build to TestFlight (App: #{app.apple_id})...".green
 
-      package_path = PackageBuilder.new.generate(apple_id: app.apple_id,
-                                                 ipa_path: config[:ipa],
-                                             package_path: "/tmp")
+      plist = FastlaneCore::IpaFileAnalyser.fetch_info_plist_file(config[:ipa])
+      package_path = FastlaneCore::IpaUploadPackageBuilder.new.generate(app_id: app.apple_id,
+                                                                      ipa_path: config[:ipa],
+                                                                  package_path: "/tmp",
+                                                                      platform: plist["DTPlatformName"])
```
Comment (KrauseFx) [inline lib/pilot/build_manager.rb]: Would it make sense to fallback to the iOS as the default platform here? As far as I know, `plist` might be `nil` if the plist file can't be found, or if there are multiple plist files and the wrong one is taken. This should not crash if `plist` is nil here

**Comment by KrauseFx** [comment]
Awesome, thanks for updating. I added one last comment :+1:

File: lib/pilot/build_manager.rb (hunk 3 above)
Comment (tmm1) [inline lib/pilot/build_manager.rb]: Fallback already exists in the method being invoked: https://github.com/fastlane/fastlane_core/blob/master/lib/fastlane_core/ipa_upload_package_builder.rb#L24

Comment (tmm1) [inline lib/pilot/build_manager.rb]: Updated to make sure it doesn't crash if plist is nil.

**Comment by KrauseFx** [comment]
This is perfect, thank you for updating the PR and contributing to `fastlane`. This will be in the next release :+1:

**Pull request merged**